### PR TITLE
Bigquery collector

### PR DESCRIPTION
**Pull request by andrewm4894**
##### Summary

Add a BigQuery collector.

##### Test Plan

Testing it using our own BigQuery creds and the random data examples in the default config.

Working as expected with multiple jobs where each job can run one or more sql queries and so have 1 or more charts.

image

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

File: collectors/python.d.plugin/bigquery/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+-->
+
+# BigQuery Netdata Collector
+
+A python based collector leveraging [`pandas-gbq`](https://pandas-gbq.readthedocs.io/en/latest/) to run a query 
+against [BigQuery](https://cloud.google.com/bigquery) and save results into a chart in Netdata.
+
+## Requirements
+
+This collector depends on some Python packages that can usually be installed via `pip`.
```
Comment (thiagoftsm) [inline collectors/python.d.plugin/bigquery/README.md:13]: Nowadays we still have have `pip` and `pip3` on hosts, my initial understanding, and probably I am wrong, this collector needs `python 2`. @Ferroin am I missing something with these `pip` versions?

File: collectors/python.d.plugin/bigquery/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+```bash
+sudo pip install google-auth pandas-gbq
```
Comment (thiagoftsm) [inline collectors/python.d.plugin/bigquery/README.md:16]: please, add a `$` in front bash

Comment (andrewm4894) [inline collectors/python.d.plugin/bigquery/README.md:16]:
```suggestion
$ sudo pip install google-auth pandas-gbq
```

File: collectors/python.d.plugin/bigquery/README.md (hunk 1 above)
Comment (andrewm4894) [reply collectors/python.d.plugin/bigquery/README.md:13]: Looks like `pandas-gbq` is only python 3. https://pypi.org/project/pandas-gbq/

So i'll need to add something maybe to make this clearer.

Comment (andrewm4894) [inline collectors/python.d.plugin/bigquery/README.md:13]:
```suggestion
This collector depends on some Python (Python 3 only) packages that can usually be installed via `pip` or `pip3`.
```

Comment (thiagoftsm) [reply collectors/python.d.plugin/bigquery/README.md:13]: I think so, only to be sure that users won't have the confusion I had.

Comment (thiagoftsm) [reply collectors/python.d.plugin/bigquery/README.md:13]: Perfect, with this I think the PR will be ready to be merged. :) 
What do you think @Ferroin ?

**Review by thiagoftsm — APPROVED** [review]
I do not have environment to test, but my concerns were addressed. During compilation I did not have any warning and bigquery was correctly installed. LGTM!

File: collectors/python.d.plugin/bigquery/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+```bash
+$ sudo pip install google-auth pandas-gbq
+```
+
+## Credentials
+
+The collector expects to read the credentials for a 
+[service account](https://cloud.google.com/iam/docs/service-accounts) from a file defined in the collector 
+configuration (see `credentials` configuration parameter). If this is not defined then auth will fallback
```
Comment (DShreve2) [inline collectors/python.d.plugin/bigquery/README.md:23]:
```suggestion
configuration (see `credentials` configuration parameter). If this is not defined, then auth will fall back 
```

**Review by DShreve2 — COMMENTED** [review]
Readme is good with one small change.

**Review by thiagoftsm — APPROVED** [review]
Last commit change only Readme.md, so LGTM!

**Comment by ilyam8** [comment]
If I understand it correctly, we need a generic SQL collector. Let's discuss it on Mon @andrewm4894

**Comment by andrewm4894** [comment]
Maybe some more generic solution using jdbc drivers and stuff like that could work and maybe be more easily extended to others.

https://cloud.google.com/bigquery/docs/reference/odbc-jdbc-drivers

**Comment by andrewm4894** [comment]
@ilyam8 i made a more general feature request thats maybe related to this. 

https://github.com/netdata/netdata/issues/13659

**Comment by andrewm4894** [comment]
@ilyam8 i think we should merge this while we wait on a generic sql collector solution. Thoughts?

File: collectors/python.d.plugin/bigquery/README.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+## Credentials
+
+The collector expects to read the credentials for a 
+[service account](https://cloud.google.com/iam/docs/service-accounts) from a file defined in the collector 
+configuration (see `credentials` configuration parameter). If this is not defined, then auth will fall back 
+to the `pandas-gbq` [approach](https://pandas-gbq.readthedocs.io/en/latest/howto/authentication.html). 
+
+## Configuration
+
+Below is an example configuration to just query some random data from BigQuery (two times) and plot each
```
Comment (DShreve2) [inline collectors/python.d.plugin/bigquery/README.md:28]:
```suggestion
Below is an example configuration to query some random data from BigQuery (two times) and plot each 
```

**Review by DShreve2 — APPROVED** [review]
Readme is good! One small comment.

**Pull request closed without merging**